File: tests/machine/x/java/user_type_literal.java
import java.util.*;
class Person {
	String name;
	int age;
	Person(String name, int age) {
		this.name = name;
		this.age = age;
	}
	@Override public boolean equals(Object o) {
		if (this == o) return true;
		if (!(o instanceof Person other)) return false;
		return Objects.equals(this.name, other.name) && Objects.equals(this.age, other.age);
	}
	@Override public int hashCode() {
		return Objects.hash(name, age);
	}
}
class Book {
	String title;
	Person author;
	Book(String title, Person author) {
		this.title = title;
		this.author = author;
	}
	@Override public boolean equals(Object o) {
		if (this == o) return true;
		if (!(o instanceof Book other)) return false;
		return Objects.equals(this.title, other.title) && Objects.equals(this.author, other.author);
	}
	@Override public int hashCode() {
		return Objects.hash(title, author);
	}
}
public class Main {
	static Book book = new Book("Go", new Person("Bob", 42));
	static <K,V> LinkedHashMap<K,V> mapOf(Object... kv) {
		LinkedHashMap<K,V> m = new LinkedHashMap<>();
		for (int i = 0; i < kv.length; i += 2) m.put((K)kv[i], (V)kv[i+1]);
		return m;
	}
	public static void main(String[] args) {
	System.out.println(book.author.name);
	}
}

File: tests/machine/x/c/user_type_literal.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Person Person;
typedef struct Book Book;

typedef struct Person {
  char *name;
  int age;
} Person;
typedef struct {
  int len;
  Person *data;
} list_Person;
static list_Person list_Person_create(int len) {
  list_Person l;
  l.len = len;
  l.data = calloc(len, sizeof(Person));
  if (!l.data && len > 0) {
    fprintf(stderr, "alloc failed\n");
    exit(1);
  }
  return l;
}

typedef struct Book {
  char *title;
  Person author;
} Book;
typedef struct {
  int len;
  Book *data;
} list_Book;
static list_Book list_Book_create(int len) {
  list_Book l;
  l.len = len;
  l.data = calloc(len, sizeof(Book));
  if (!l.data && len > 0) {
    fprintf(stderr, "alloc failed\n");
    exit(1);
  }
  return l;
}

int main() {
  Book book =
      (Book){.title = "Go", .author = (Person){.name = "Bob", .age = 42}};
  printf("%s\n", book.author.name);
  return 0;
}
